### ISSUE TITLE
efficient implementation 

### ISSUE THREAD

**Issue by amrhamedp** [lucidrains/En-transformer#5]
Hi,
I wonder if relative distances and coordinates can be handled more efficiently using memory efficient attention as in " Self-attention Does Not Need O(n^2) Memory". It is straightforward for the scalar part.

**Comment by amrhamedp** [comment]
yes ...it is pretty useful however I believe that distances should be used to allow nodes positional encoding. I mean something like IPA (invariant point attention) or schnet  like encoding for distances using RBF.

**Comment by lucidrains** [comment]
@amrhamedp they do though, the distance information is rotated into the keys, which is propagated to both the nodes and coordinates
